src/util/base58.rs

```diff
@@ -36,8 +36,8 @@ pub enum Error {
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
-            Error::BadByte(b) => write!(f, "invalid base58 character 0x{:x}", b),
-            Error::BadChecksum(exp, actual) => write!(f, "base58ck checksum 0x{:x} does not match expected 0x{:x}", actual, exp),
+            Error::BadByte(b) => write!(f, "invalid base58 character {:#x}", b),
+            Error::BadChecksum(exp, actual) => write!(f, "base58ck checksum {:#x} does not match expected {:#x}", actual, exp),
```

Comment:
TBH I'm not sure if/why this is better but I don't mind.

Reply:
I thought it would be better to use `#` since we are making an effort to correctly handle alternate in our `Display` impls now.

Reply:
FTR I only learned about alternate a month or so ago when you prompted me to implement it :)